### PR TITLE
fix library issue by impl catch-all logics

### PR DESCRIPTION
**Pull request by kaibocai**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [X] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [X] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [X] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [X] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by amamounelsayed — DISMISSED** [review]

File: src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java [hunk 1]
```diff
@@ -30,18 +30,23 @@ public ClassLoader createClassLoader() {
     return classLoader;
   }
 
-  @Override
-  public void addUrl(URL url) throws IOException {
+  public void addUrl(URL url, String source) throws IOException {
```
Comment (shreyas-gopalakrishna) [inline src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java]:
```suggestion
  public void addUrl(URL url) throws IOException {
```
Please let us check if source is necessary

File: src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java [hunk 2]
```diff
@@ -30,18 +30,23 @@ public ClassLoader createClassLoader() {
...
-    WorkerLogManager.getSystemLogger().info("Loading file URL: " + url);    
-
+    WorkerLogManager.getSystemLogger().info("Loading " + source + " file URL: " + url);
     urls.add(url);
+    addUrlToSystemClassLoader(url);
+  }
 
+  @Override
+  public void addCustomerUrl(URL url) throws IOException {
+    addUrl(url, "customer");
```
Comment (shreyas-gopalakrishna) [inline src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java]:
```suggestion
    addUrl(url);
```
We can review if source is needed

File: src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java [hunk 3]
```diff
@@ -30,18 +30,23 @@ public ClassLoader createClassLoader() {
...
+  }
 
-    addUrlToSystemClassLoader(url);
+  @Override
+  public void addWorkerUrl(URL url) throws IOException {
+    addUrl(url, "worker");
```
Comment (shreyas-gopalakrishna) [inline src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java]:
```suggestion
    addUrl(url);
```

**Review by shreyas-gopalakrishna — COMMENTED** [review]
Looks good. Let us add unit/end-to-end tests to be safe

File: src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java (hunk 1 above)
Comment (trask) [reply src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java]:
```suggestion
  private void addUrl(URL url, String source) throws IOException {
```

File: src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java (hunk 2 above)
Comment (trask) [inline src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java]: I think two different underlying lists would be less fragile

File: src/main/java/com/microsoft/azure/functions/worker/reflect/EnhancedClassLoaderProvider.java [hunk 4]
```diff
@@ -11,7 +11,8 @@
 
 public class EnhancedClassLoaderProvider implements ClassLoaderProvider {
```
Comment (trask) [inline src/main/java/com/microsoft/azure/functions/worker/reflect/EnhancedClassLoaderProvider.java:12]:
```suggestion
// Java 11
public class EnhancedClassLoaderProvider implements ClassLoaderProvider {
```

File: src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java [hunk 5]
```diff
@@ -13,35 +13,43 @@
...
-  public ClassLoader createClassLoader() {
-    URL[] urlsForClassLoader = new URL[urls.size()];
-    urls.toArray(urlsForClassLoader);
+  public ClassLoader createClassLoader() throws IOException {
+    List<URL> urlsList = new ArrayList<>();
+    urlsList.addAll(this.customerUrls);
+    urlsList.addAll(this.workerUrls);
+    addUrlToSystemClassLoader(urlsList);
 
+    //TODO: validate below lines should not exist
```
Comment (trask) [inline src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java]: a little more context to this TODO might be helpful

**Review by trask — DISMISSED** [review]

File: src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java (hunk 5 above)
Comment (kaibocai) [reply src/main/java/com/microsoft/azure/functions/worker/reflect/DefaultClassLoaderProvider.java]: right! will add more context to this TODO item.

**Review by amamounelsayed — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Review by shreyas-gopalakrishna — APPROVED** [review]

**Pull request merged**